client/bgp_instance_test.go

```diff
@@ -0,0 +1,125 @@
+		RedistributeOtherBgp:     redistributeOtherBgp,
+		RedistributeRip:          redistributeRip,
+		RedistributeStatic:       redistributeStatic,
+		RouterID:                 routerID,
+		RoutingTable:             routingTable,
+	}
+	bgpInstance, err := c.AddBgpInstance(expectedBgpInstance)
+	if err != nil {
+		t.Fatalf("Error creating a bpg instance with: %v", err)
+	}
+
+	expectedBgpInstance.ID = bgpInstance.ID
+
+	if !reflect.DeepEqual(bgpInstance, expectedBgpInstance) {
+		t.Errorf("The bgp instance does not match what we expected. actual: %v expected: %v", bgpInstance, expectedBgpInstance)
+	}
+
+	err = c.DeleteBgpInstance(bgpInstance.Name)
+
+	if err != nil {
+		t.Errorf("Error deleting bgp instance with: %v", err)
+	}
+}
+
+func TestAddAndUpdateBgpInstanceWithOptionalFieldsAndDeleteBgpInstance(t *testing.T) {
+	c := NewClient(GetConfigFromEnv())
+
+	expectedBgpInstance := &BgpInstance{
+		Name:                     bgpName,
+		As:                       as,
+		ClientToClientReflection: clientToClientReflection,
+		Comment:                  bgpComment,
+		ConfederationPeers:       confederationPeers,
+		Disabled:                 disabled,
+		IgnoreAsPathLen:          ignoreAsPathLen,
+		OutFilter:                outFilter,
+		RedistributeConnected:    redistributeConnected,
+		RedistributeOspf:         redistributeOspf,
+		RedistributeOtherBgp:     redistributeOtherBgp,
+		RedistributeRip:          redistributeRip,
+		RedistributeStatic:       redistributeStatic,
+		RouterID:                 routerID,
+		RoutingTable:             routingTable,
+		ClusterID:                clusterID,
+		Confederation:            confederation,
+	}
+	bgpInstance, err := c.AddBgpInstance(expectedBgpInstance)
+	if err != nil {
+		t.Fatalf("Error creating a bpg instance with: %v", err)
+	}
+
+	expectedBgpInstance.ID = bgpInstance.ID
+
+	if !reflect.DeepEqual(bgpInstance, expectedBgpInstance) {
+		t.Errorf("The bgp instance does not match what we expected. actual: %v expected: %v", bgpInstance, expectedBgpInstance)
+	}
+
+	// update fields
+	expectedBgpInstance.Confederation = updatedConfederation
+	expectedBgpInstance.As = updatedAs
+
+	bgpInstance, err = c.UpdateBgpInstance(expectedBgpInstance)
+
+	if !reflect.DeepEqual(bgpInstance, expectedBgpInstance) {
+		t.Errorf("The bgp instance does not match what we expected. actual: %v expected: %v", bgpInstance, expectedBgpInstance)
+	}
+
+	err = c.DeleteBgpInstance(bgpInstance.Name)
+
+	if err != nil {
+		t.Errorf("Error deleting bgp instance with: %v", err)
+	}
+}
+
+func TestFindBgpInstance_onNonExistantBgpInstance(t *testing.T) {
+	c := NewClient(GetConfigFromEnv())
+
+	name := "bgp instance does not exist"
+	_, err := c.FindBgpInstance(name)
+
+	expectedErrStr := fmt.Sprintf("bgp instance `%s` not found", name)
+	if err == nil || err.Error() != expectedErrStr {
+		t.Errorf("client should have received error indicating the following bgp instance `%s` was not found. Instead error was nil", name)
+	}
```

Comment:
I think it would be cleaner to just check the error type here since it's exported.

Something like this

```
_, err := c.FindBgpInstance(name)

if _, ok := err.(NotFound); !ok {
  t.Errorf("expecting to receive NotFound error for bgp instance `%s`, instead error was nil", name)
}
```